backend/globaleaks/handlers/auth/__init__.py

```diff
@@ -97,8 +115,14 @@ def login_whistleblower(session, tid, receipt, client_using_tor, operator_id=Non
     return session
 
 
+def check_user_fiscal_code(user, fiscal_code, tid):
+    if tid == 1:
+        return True
+    if user.fiscal_code and user.fiscal_code != fiscal_code:
+        raise errors.ForbiddenOperation
```

Comment:
SECURITY ISSUE:  An equality check of this type is prone to tempering. 

For this reason we have specified to integrate the IDP using signature verification of the token.

This will need to be reimplemented integrating the code we prototyped in: https://github.com/globaleaks/globaleaks-whistleblowing-software/issues/4107#issuecomment-2171796635